src/plugins/action_generate_review_assignment_records/src/databaseMethods.ts

```diff
@@ -0,0 +1,131 @@
+        reviewerId,
+        orgId,
+        stageId,
+        stageNumber,
+        status,
+        applicationId,
+        templateSectionRestrictions,
+        level,
+        isLastLevel,
+      } = reviewAssignment
+      // Needs a slightly different query with different CONFLICT restrictions
+      // depending on whether orgId exists or not.
+      // On conflict, existing records have their Section Restrictions updated,
+      // but assignment status remains unchanged.
+      const text = `
+        INSERT INTO review_assignment (
+          reviewer_id, stage_id,
+          stage_number, status, application_id,
+          template_section_restrictions, level, is_last_level
+          ${orgId ? ', organisation_id' : ''}
+          )
+        VALUES ($1, $2, $3, $4, $5, $6, $7, $8${orgId ? ', $9' : ''})
+        ON CONFLICT (reviewer_id,${
+          orgId ? ' organisation_id,' : ''
+        } stage_number, application_id, level)
+          WHERE organisation_id IS ${orgId ? 'NOT ' : ''}NULL
+        DO
+          UPDATE SET template_section_restrictions = $6
+        RETURNING id`
+
+      try {
+        const result = await DBConnect.query({
+          text,
+          values: [
+            reviewerId,
+            stageId,
+            stageNumber,
+            status,
+            applicationId,
+            templateSectionRestrictions,
+            level,
+            isLastLevel,
+            ...(orgId ? orgId : []),
+          ],
+        })
+        reviewAssignmentIds.push(result.rows[0].id)
+
+        // TO-DO: What to do with existing records that don't match the
+        // generated ones? Delete them? Set their status = "Not Available"?
+      } catch (err) {
+        console.log(err.message)
+        reviewAssignmentIds.push(err.message)
+        throw err
+      }
+    }
+    return reviewAssignmentIds
+  },
+  addReviewAssignmentAssignerJoins: async (reviewAssignmentAssignerJoins: any) => {
+    const reviewAssignmentAssignerJoinIds = []
+    for (const reviewAssignmentAssignerJoin of reviewAssignmentAssignerJoins) {
+      const { assignerId, orgId, reviewAssignmentId } = reviewAssignmentAssignerJoin
+      // Needs a slightly different query with different CONFLICT restrictions
+      // depending on whether orgId exists or not.
+      const text = `
+        INSERT INTO review_assignment_assigner_join (
+          assigner_id,
+          review_assignment_id
+          ${orgId ? ', organisation_id' : ''}          
+          )
+        VALUES ($1, $2${orgId ? ', $9' : ''})
+        ON CONFLICT (assigner_id,
+          review_assignment_id
+          ${orgId ? ', organisation_id' : ''} 
+        )
+          WHERE organisation_id IS ${orgId ? 'NOT ' : ''}NULL
+        DO UPDATE SET organisation_id = ${orgId ? 'orgId' : 'NULL'}
+        RETURNING id
+      `
+      try {
+        const result = await DBConnect.query({
+          text,
+          values: [assignerId, reviewAssignmentId, ...(orgId ? orgId : [])],
```

Comment:
I think this needs brackets around orgId
`...(orgId ? [orgId]: [])`

also I think you can have orgId in the values list and have `VALUES ($1, $2, $3)`

Reply:
> I think this needs brackets around orgId
> `...(orgId ? [orgId]: [])`

Yeah I think you're right. Which is weird cos it's copied and pasted from the previous one, so I guess that's been wrong too. Guess we've had no users with orgs being added. Probably should write some tests for that...

> also I think you can have orgId in the values list and have `VALUES ($1, $2, $3)`

Oh yeah, that works. Cool. Still have to use the `orgId ? ' organisation_id,' : ''` for the conflict resolution though.

Have changed both methods now: 467704d9